### PR TITLE
Fix issue with empty string columns

### PR DESCRIPTION
**Pull request by Jarek-Rolski**

### CODE REVIEW

File: api/api/application/services/dataset_validation.py [hunk 1]
```diff
@@ -106,6 +106,9 @@ def dataset_has_correct_data_types(
         data_frame,
     )
     for column in schema.columns:
+        if column.name not in column_types:
```
Comment (TobyDrane) [inline api/api/application/services/dataset_validation.py:109]: I don't understand why we are matching the column name with the column type? Wouldn't we expect the column datatype to match the column_type?

If we then make that change this function would be a bit redundant as every column would always pass.

**Review by TobyDrane — COMMENTED** [review]
I've tested this locally with the csv file you originally uploaded in the issue and I can't see to bypass the error with this code pulled locally, is it working as expected?

When uploading the csv via the create schema it is still inferring the empty string column  as a double sadly.

File: api/api/application/services/dataset_validation.py (hunk 1 above)
Comment (Jarek-Rolski) [reply api/api/application/services/dataset_validation.py:109]: I check here if column.name key is in column_types dictionary. I modified extract_athena_types function to not include that key if column is empty.

Comment (Jarek-Rolski) [reply api/api/application/services/dataset_validation.py:109]: I used API to test it. I first create schema and then upload it.
If you try to create schema from dataframe with empty column, it by default infers double, I guess.

**Comment by Jarek-Rolski** [comment]
My original bug fix didn't work because I tested it for empty columns with None values it causes column dtype to be object and "infer_dtype" function interprets series dtype from its values. Upload with UI sets null values to np.nan, it causes data frame column dtype to be float and "infer_dtype" function interprets float series dtype from it's dtype not values. 
I additionally fixed and added checks to test_return_error_message_when_not_correct_datatypes test function.

**Comment by MotwaniM** [comment]
Hi, we're looking to merge this into our next release. This looks good to me.

**Pull request merged**